clang/lib/Sema/SemaOpenMP.cpp

```diff
@@ -19520,6 +19520,13 @@ static bool actOnOMPReductionKindClause(
   bool FirstIter = true;
   for (Expr *RefExpr : VarList) {
     assert(RefExpr && "nullptr expr in OpenMP reduction clause.");
+    if (ClauseKind == OMPC_reduction &&
+        RD.RedModifier == OMPC_REDUCTION_inscan && RefExpr->isTypeDependent()) {
+      S.Diag(RefExpr->getExprLoc(),
+             diag::err_omp_inscan_reduction_on_template_type);
+      continue;
+    }
+
```

Comment:
This is definetely wrong, templates should be supported

Reply:
It should ideally be supported. However, the compiler crashes as per #67002 on the latest builds.

Reply:
> This is definetely wrong, templates should be supported

Did you mean that templates are currently supported, or did you mean that they should be supported?

Reply:
They are supported, but this check disables this support